.github/workflows/release.yaml

```diff
@@ -0,0 +1,101 @@
+name: "release"
+
+on:
+  push:
+    branches: ["main"]
+
+env:
+  G_DIST_ARTIFACT_NAME: "yscope-log-viewer-dist-${{github.sha}}"
+
+permissions: {}
+
+concurrency:
+  group: "${{github.workflow}}-${{github.ref}}"
+
+  # Cancel in-progress jobs for efficiency
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "actions/checkout@v4"
+        with:
+          persist-credentials: false
+          submodules: "recursive"
+
+      - uses: "actions/setup-node@v4"
+        with:
+          node-version: "22"
+          cache: "npm"
+          cache-dependency-path: "./package-lock.json"
+
+      - run: "npm install"
+
+      - run: "npm run build"
+
+      - name: "Upload static files as artifact"
+        uses: "actions/upload-pages-artifact@v3"
+        with:
+          name: "${{env.G_DIST_ARTIFACT_NAME}}"
+          path: "./dist"
+
+  release:
+    needs: "build"
+    runs-on: "ubuntu-latest"
+    permissions:
+      # To create a release
+      contents: "write"
+    steps:
+      - uses: "actions/checkout@v4"
+        with:
+          persist-credentials: false
+          submodules: "recursive"
+
+      - name: "Determine release names"
+        id: "names"
+        run: |
+          PACKAGE_VERSION=$(jq -r ".version" "package.json")
+          BRANCH_NAME=$(echo "${GITHUB_REF}" | sed 's|refs/heads/||')
+          DATE=$(date +'%Y%m%d')
+          SHORT_SHA=$(git rev-parse --short HEAD)
+          VERSION=${PACKAGE_VERSION}-${BRANCH_NAME}+${DATE}.${SHORT_SHA}
+
+          echo "VERSION=${VERSION}" >> "$GITHUB_OUTPUT"
+          echo "DIST_TAR_NAME=dist-${VERSION}.tar" >> "$GITHUB_OUTPUT"
+
+      - uses: "actions/download-artifact@v4"
+        with:
+          name: "${{env.G_DIST_ARTIFACT_NAME}}"
+
+      - name: "Rename build artifact"
```

Comment:
alternatively, in the `build` job directly tar the dist with compression, so we don't have to rely on "actions/upload-pages-artifact@v3" to tar the dist.

Reply:
e.g.,
```
          tar --create \
          --gzip \
          --file "${DIST_TAR_NAME}" \
          --transform="s|^dist|${VERSIONED_DIST_NAME}|" \
          dist
```

Reply:
Is there a strong reason to not use "upload-pages-artifact" ?